src/main/java/htsjdk/samtools/util/AbstractLocusIterator.java

```diff
@@ -0,0 +1,572 @@
+     * If false, emit a AbstractLocusInfo only if a locus has coverage.
+     */
+    private boolean emitUncoveredLoci = true;
+
+    /**
+     * If set, this will cap the number of reads we accumulate for any given position.
+     * Note that if we hit the maximum threshold at the first position in the accumulation queue,
+     * then we throw further reads overlapping that position completely away (including for subsequent positions).
+     * This is a useful feature if one wants to minimize the memory footprint in files with a few massively large pileups,
+     * but it must be pointed out that it could cause major bias because of the non-random nature with which the cap is
+     * applied (the first maxReadsToAccumulatePerLocus reads are kept and all subsequent ones are dropped).
+     */
+    private int maxReadsToAccumulatePerLocus = Integer.MAX_VALUE;
+
+    /**
+     * Set to true when we have enforced the accumulation limit for the first time
+     */
+    private boolean enforcedAccumulationLimit = false;
+
+    /**
+     * If true, include indels in the LocusInfo
+     */
+    protected boolean includeIndels = false;
+
+    /**
+     * When there is a target mask, these members remember the last locus for which a AbstractLocusInfo has been
+     * returned, so that any uncovered locus in the target mask can be covered by a 0-coverage AbstractLocusInfo
+     */
+    private int lastReferenceSequence = 0;
+
+    /**
+     * Last processed locus position in the reference
+     */
+    private int lastPosition = 0;
+
+    /**
+     * Set to true when past all aligned reads in input SAM file
+     */
+    private boolean finishedAlignedReads = false;
+
+    private final LocusComparator<Locus> locusComparator = new LocusComparator<>();
+
+    /**
+     * Last processed interval, relevant only if list of intervals is defined.
+     */
+    private int lastInterval = 0;
+
+    /**
+     * Prepare to iterate through the given SAM records, skipping non-primary alignments
+     *
+     * @param samReader    must be coordinate sorted
+     * @param intervalList Either the list of desired intervals, or null.  Note that if an intervalList is
+     *                     passed in that is not coordinate sorted, it will eventually be coordinated sorted by this class.
+     * @param useIndex     If true, do indexed lookup to improve performance.  Not relevant if intervalList == null.
+     *                     It is no longer the case the useIndex==true can make performance worse.  It should always perform at least
+     *                     as well as useIndex==false, and generally will be much faster.
+     */
+
+    public AbstractLocusIterator(final SamReader samReader, final IntervalList intervalList, final boolean useIndex) {
+        final String className = this.getClass().getSimpleName();
+        if (samReader.getFileHeader().getSortOrder() == null || samReader.getFileHeader().getSortOrder() == SAMFileHeader.SortOrder.unsorted) {
```

Comment:
second part of predicate should be 
samReader.getFileHeader().getSortOrder() != SAMFileHeader.SortOrder.coordinate

rather than ==unsorted, since order could also be queryname and that would also be bad.